tests/workingWithAPI.spec.ts

```diff
@@ -26,15 +26,20 @@ test('mocking API request', async ({page}) => {
 
     await pm.onOwnersPage().validateOwnerCount(2)
 
-    // Capturing data from the testOwners JSON file, which contains information to be displayed on the Owners Page
-    const testOwnerFullName = `${testOwners[0]['firstName']} ${testOwners[0]['lastName']}`
-    
+    // Extracting full name from API Mock to click on link in 
+    const testOwnerFullName = `${testOwners[0]['firstName']} ${testOwners[0]['lastName']}`    
     await pm.onOwnersPage().clickOwnerNameFor(testOwnerFullName)
     
-    // Specific owner information from the testOwners file should be displayed on the Owner Information page
     await pm.onOwnerInformationPage().validateOwnerInformation(testOwners[0]['firstName'], testOwners[0]['lastName'], testOwners[0]['address'], testOwners[0]['city'], testOwners[0]['telephone'])
 
-    await pm.onOwnerInformationPage().validatePetNamesArePresentFromListOfPetNames([testOwners[0]['pets'][0]['name'], testOwners[0]['pets'][1]['name']])
+    // Extracting all pet names from API Mock into a list
+    const petNameList: string[] = []
+    for(let pet of testOwners[0]['pets']){
+        petNameList.push(pet['name'])
+    }
```

Comment:
Do you want to hide all this soring logic into the method responsible for the validation of pets?
You just need to provide the pets as an argument, you expect to validate, and the method should do "all the magic"

The body of the test should have remained clean with just clear and understandable test steps. And all implementation, of how this done, should be hidden inside of the step.